### ISSUE TITLE
Recurring all-day events are not showed today

### ISSUE THREAD

**Issue by jmcortesmartinez** [andstatus/todoagenda#89]
If you have a recurring all-day event you can see it all the others days except today.
It does not affect whether the event is new or old.

I have detected this issue in the last few days. Before it worked properly.

Thank you for your work!

**Comment by ls5454ss** [comment]
I am having this same issue.  I have to enable one day old events to see them, and they are showing as "old/past due"...even though the date is today's date.  This is frustrating, as I'm stuck seeing ACTUAL old events as well, which I do not want to see.

Is this project still being worked?  Is this something that will be resolved?  Thank you for your time

- Steve

**Comment by jmcortesmartinez** [comment]
Maybe this captures will help you, despite being in spanish:

1.- This is a capture of my Google Calendar widget showing birthdays in March

![Screenshot_20220309-180804](https://user-images.githubusercontent.com/7203044/157493976-baed2778-db65-4e64-a8bf-a3525c57836e.png)

2.- And this is a capture of my TodoAgenda widget

![Screenshot_20220309-180820](https://user-images.githubusercontent.com/7203044/157494289-3af6f549-8a0a-4387-b1ac-a5a8426fa02a.png)

As you can see today is the birthday of Luis but I can't see it in the TodoAgenda widget.

Thank you very much for you time,
Jose

**Comment by yvolk** [comment]
@jmcortesmartinez In order to fix the problem please try this:
In the widget's settings go to "Event Filters"
then scroll down to "Filters troubleshooting"
and set it to "Debug filtering".

Today's recurring events should appear.

?!

**Comment by yvolk** [comment]
@jmcortesmartinez @ls5454ss I made a fix and posted v.4.5.0 here: #3
Please check.

**Comment by jmcortesmartinez** [comment]
Sorry, the v.4.5.0 closes instantly when opened.

I have installed it from Files with the apk.

I have a Pixel 3a with Android 12
Last security update: Feb-5 202

**Comment by ls5454ss** [comment]
@yvolk I'm not sure if I'm doing anything wrong here, but, the new version doesn't seem to be working at all.  I had to uninstall 4.4 to get 4.5 to install (said there was a conflict when trying to upgrade).  I was able to install the new version, but when trying to customize the widget, it crashed.  Now when I try to click the "blank" widget, the app immediately crashes.  Unsure if it's something on my end or not.  Uninstalling/reinstalling does not seem to work, nor does restarting the phone.

Reinstalling 4.4 seems to work.  I'm on a Pixel 6 Pro if that helps.  Android 12.  Thanks for your assistance

**Comment by yvolk** [comment]
@ls5454ss Thank you for checking.
I see that the problem with v.4.5.0 was due to specific to new requirements for Android 12.
I fixed the problem, checked it on my Android 12 device - it works there, and published v.4.5.1.
Please check it. https://github.com/andstatus/todoagenda/issues/3

**Comment by jmcortesmartinez** [comment]
It works!

Thank you very much for your time and your work. Impressive.

**Comment by ls5454ss** [comment]
@yvolk it seems to work so far - thank you!  I'll let you know via this thread if I see any more issues in regards to these events not showing.  Thank you so much for your time!

The only random issue I see, which MAY have been seen in a previous version, is that some of my EXTREMELY old recurring events (some could even be 10 years old...) are shown as a "darker gray".  Screenshot attached comparing a very old version of the widget to this latest version.  Latest version is on the top.  I honestly don't mind just re-creating those very old recurring events if needed...but just thought I would let you know.

Again - thank you for your time and talents :)

- Steve 

![Todo Screenshot](https://user-images.githubusercontent.com/101206895/157683264-72088f21-4344-42da-badf-ecb206c68471.png)

**Comment by ls5454ss** [comment]
@yvolk it seems to work so far - thank you! I'll let you know via this thread if I see any more issues in regards to these events not showing. Thank you so much for your time!

:: edit ::
I do show one issue (I think it's an issue...). When I click any of the events, they do not load in Google Calendar - they do nothing. Unsure if this is related to whatever changes you may have just made

The second random issue I see, which MAY have been seen in a previous version, is that some of my EXTREMELY old recurring events (some could even be 10 years old...) are shown as a "darker gray". Screenshot attached comparing a very old version of the widget to this latest version. Latest version is on the top. I honestly don't mind just re-creating those very old recurring events if needed...but just thought I would let you know.

Again - thank you for your time and talents :)

![Todo Screenshot](https://user-images.githubusercontent.com/101206895/157683785-5917f3b9-b717-4d2f-9043-ee394e067f31.png)

**Comment by ls5454ss** [comment]
> > I do show one issue (I think it's an issue...). When I click any of the events, they do not load in Google Calendar - they do nothing. Unsure if this is related to whatever changes you may have just made
> 
> Yes, this is a new bug. Will fix it now.

Thank you for your time, @yvolk - much appreciated!

**Comment by yvolk** [comment]
> The second random issue I see, which MAY have been seen in a previous version, is that some of my EXTREMELY old recurring events (some could even be 10 years old...) are shown as a "darker gray". Screenshot attached comparing a very old version of the widget to this latest version. Latest version is on the top. I honestly don't mind just re-creating those very old recurring events if needed...but just thought I would let you know.

This issue / feature may be related to the recent change "Implement strikethrough and partial transparency for event status" see #82. In a case that new feature works not as intended, please create a separate issue for it.

**Comment by yvolk** [comment]
@ls5454ss I fixed the bug with not working clicks on events in Android 12: here the "PendingIntent" needed to be "MUTABLE": 
I fixed this also, please check v.4.5.2 here: #3

**Comment by ls5454ss** [comment]
> @ls5454ss I fixed the bug with not working clicks on events in Android 12: here the "PendingIntent" needed to be "MUTABLE", please check v.4.5.2 here: #3

@yvolk unfortunately it doesn't seem to be working...sorry.  Similar issue to the one above where the app would immediately crash / force close.  It seems to install, let you grant permissions to Calendar, go in to the customize screen, and that's about it.  Once you close the customize screen, the widget is not shown, and the app will keep force closing after that.

**Comment by ls5454ss** [comment]
> @ls5454ss I fixed the bug with not working clicks on events in Android 12: here the "PendingIntent" needed to be "MUTABLE", please check v.4.5.2 here: #3

That seems to have done it - thank you once again for your time!  :)

- Steve

**Comment by yvolk** [comment]
@ls5454ss @jmcortesmartinez  Thank you for your feedback!

**Comment by ls5454ss** [comment]
Hi @yvolk  - sorry to be the bearer of bad news, but something is still not right.  The day is not over yet ~7:00pm, and my event is now not showing in the widget.  Any idea what could be wrong, with the widget thinking the event is already done, even though the day isn't over?  Two screenshots shown - one of the actual event, one of the widget VS a Google Calendar widget.

Thanks again for your time
![1](https://user-images.githubusercontent.com/101206895/157777459-f3ea244f-f1fc-433e-987a-970a0ba0f3ad.png)
![2](https://user-images.githubusercontent.com/101206895/157777461-6dac5247-b732-4451-8a59-8a9a2cdb55a6.png)
!

**Comment by yvolk** [comment]
@ls5454ss 
Q1. Do I understand correctly that before my changes similar events disappeared at midnight on the day of the all day event, and now they disappear 19 hours later than before?!
What I actually did: I moved earliest selection date-time one full day back, so I would expect that all events will disappear one full day later.

Q2. Please in "Event filters" allow "Show past events: **Ended today**" - do you see the Today's All day event? Is it shown as Today in the widget?

Q3. Please in "Event filters" allow "Show past events: **Ended yesterday**" - do you see the Today's All day event? Is it shown as Today in the widget?

**Comment by ls5454ss** [comment]
Hi @yvolk 

Q1:  I am not sure.  I've been using a VERY old version of this, considering when I tried the latest version (before we started talking) it didn't work either.  I'd have to install version 4.4.0 and test to answer this question.  If you'd like me to, I will, and compare between 4.4.0 and 4.5.2.  I do recall, even on my very old version, that SOME all day recurring events would show initially, and remove before the end of the day.  I say this, because I usually "pay my bills" at the end of the day, and missed a few because they weren't showing when I looked at the widget. 

Note:  Your "note" does make sense.  I tried messing around in the settings last night, thinking the same thing - but didn't really see anything that would help, other than trying to mess with the time zone on my phone (which I didn't get to try).

Q2.  I can try this later tonight, when my event goes away.  I have one recurring event (all day) and I created a new one (all day, recurring) as well.  Stay tuned.  

Q3.  Same as Q2 - stay tuned.  

Q4.  Sure thing - will do. 

Thank you again for your time!

**Comment by ls5454ss** [comment]
Hello again, @yvolk 

Q2 - "Event filters" allow "Show past events: Ended today" - this made the events show up.  

Q4 - file attached, compressed.  Couldn't attach a JSON

Thanks again for your time!

[Todo-Agenda-29-Todo-Agenda-shareEvents-2022-03-11-19-58-02-646.json.zip](https://github.com/andstatus/todoagenda/files/8236696/Todo-Agenda-29-Todo-Agenda-shareEvents-2022-03-11-19-58-02-646.json.zip)

**Comment by yvolk** [comment]
@ls5454ss Thank you for the info. After my investigations and attempts to reproduce the problem I think that the only problem is in the start date/time of the query, which I corrected (by one day) once.
Now I ended in 2 days correction and looks like this helps to show All day events.
Please check v.4.5.3 here #3

**Comment by ls5454ss** [comment]
@yvolk - thank you again for your time.  Will install now and provide feedback soon :)

**Comment by ls5454ss** [comment]
@yvolk I think whatever you changed seems to be working now.  I didn't have any issues with the all day events, and they seemed to clear ~ midnight now.

I'll keep you posted if I see any more issues with this.  Thank you again for your time, and have a great rest of the weekend!
